### ISSUE TITLE
unpacking a dict in function call results in type error

### ISSUE THREAD

**Issue by ehsangifani** [microsoft/pyright#6061]
## Environment data
-   Language Server version: 2023.9.30 (pyright 7e72aa15)
-   OS and version: Windows 11 (22H2)
-   Python version (& distribution if applicable, e.g. Anaconda): 2023.9.30

## Code Snippet

```python
def example(a: str, b: int = 10, c: bool = False):
    pass
example('a', **{'b': 10})
```

## Expected behavior

There should not be any type error as `b` has the correct type. And because the dictionary doesn't contain a key for `c`, there shouldn't be any problem.

## Actual behavior

```
Argument of type "int" cannot be assigned to parameter "c" of type "bool" in function "example"
  "int" is incompatible with "bool" Pylance(reportGeneralTypeIssues)
```

**Comment by erictraut** [comment]
This is a duplicate of #5486. The behavior you're seeing here is intended, even though I agree it's not ideal.

Pyright previously did not generate an error for this condition, but I changed it to match the behavior of mypy, another Python type checker. Pyright's behavior does [deviate from mypy](https://microsoft.github.io/pyright/#/mypy-comparison) in some cases, but this particular case affects overload matching, and it's important for all Python type checkers to have the same overload matching behaviors.

There is an ongoing effort to standardize Python static type checking behaviors. This effort also includes the proposed creation of a "typing council" that will provide a means for changing and formalizing behaviors like this one. If and when that happens, I will advocate for this change across type checkers. Until that time, we're stuck with the current behavior.
